app/src/server/trpc/routers/service.ts

```diff
@@ -62,13 +118,111 @@ export const serviceRouter = router({
+  draftParent: publicProcedure
     .input(z.object({ resourceType: z.enum(['Measure', 'Library']), id: z.string() }))
     .mutation(async ({ input }) => {
+      // fetch the artifact from the measure repository using its resource type and id
       const draftRes = await fetch(`${process.env.NEXT_PUBLIC_MRS_SERVER}/${input.resourceType}/${input.id}`);
-      const draftArtifact = modifyResourceToDraft((await draftRes.json()) as FhirArtifact);
+      const draftJson = (await draftRes.json()) as FhirArtifact;
+
+      // recursively get any child artifacts from the artifact if they exist
+      const children = draftJson.relatedArtifact ? await getChildren(draftJson.relatedArtifact) : [];
 
+      const draftArtifact = modifyResourceToDraft({ ...draftJson });
+
+      // create a draft of the modified parent artifact
       const res = await createDraft(input.resourceType, draftArtifact);
-      return { draftId: draftArtifact.id, ...res };
+
+      return { draftId: draftArtifact.id, children, ...res };
+    }),
+
+  draftChild: publicProcedure
+    .input(z.object({ resourceType: z.enum(['Measure', 'Library']), url: z.string(), version: z.string() }))
+    .mutation(async ({ input }) => {
+      // search the measure repository for an artifact of the same resource type, url, and version
+      // assume that artifacts will have unique urls and versions, so return the first entry in the bundle
+      const artifactBundle = await fetch(
+        `${process.env.NEXT_PUBLIC_MRS_SERVER}/${input.resourceType}?` +
+          new URLSearchParams({ url: input.url, version: input.version })
+      ).then(resArtifacts => resArtifacts.json() as Promise<fhir4.Bundle<FhirArtifact>>);
+
+      if (!artifactBundle.entry || artifactBundle.entry.length === 0) {
+        throw new Error('No artifacts found in search');
+      }
+      const draftRes = artifactBundle.entry?.[0].resource;
+
+      // increment the version in the url and update the relatedArtifact.resource to have the new version on the url
+      const draftChildArtifact = modifyResourceToDraft(draftRes as FhirArtifact);
+
+      // create a draft of the modified child artifact
+      const res = await createDraft(input.resourceType, draftChildArtifact);
+
+      return {
+        draftId: draftChildArtifact.id,
+        id: draftRes?.id,
+        resourceType: draftChildArtifact.resourceType,
+        ...res
+      };
+    }),
+
+  // new release procedures (separate release of parent and release of child)
+  releaseParent: publicProcedure
+    .input(z.object({ resourceType: z.enum(['Measure', 'Library']), id: z.string(), version: z.string() }))
+    .mutation(async ({ input }) => {
+      // get the parent draft artifact by id
+      const draftRes = await getDraftById(input.id, input.resourceType);
+
+      // if a draft artifact of that resource type and id exists, then modify its content to be an active artifact
+      if (draftRes) {
+        draftRes.version = input.version;
```

Comment:
I think out of scope for this task but maybe worth thinking about: I don't think we allow editing on the version, so theoretically we could end up releasing with a version that already exists on the Measure repo, and the user has no way of changing it.
Example: 0.0.001 and 0.0.002 exist in repo. 0.0.001 is used to create a draft, which is auto assigned version 0.0.002. 0.0.002 draft is released and collides with existing 0.0.002 in the repo.

Reply:
I totally agree that this is something we will have to address. This is documented (or at least mostly) in [this task](https://jira.mitre.org/browse/TACOSTRAT-1386), so I will hold off on this functionality if you think that's okay.

Reply:
Cool! I just added a comment to the task that "We also need released artifacts to not conflict with drafts." since the task seems to just be enforcing unique urls across drafts.